sklearn/utils/tests/test_stats.py

```diff
@@ -96,3 +96,71 @@ def test_weighted_percentile_2d():
         _weighted_percentile(x_2d[:, i], w_2d[:, i]) for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
+
+
+def test_weighted_percentile_nan_filtered():
+    """Test that calling _weighted_percentile on an array with nan values returns
+    the same results as calling _weighted_percentile on a filtered version of the data.
+    We test both with sample_weight of the same shape as the data and for
+    one-dimensional sample_weight."""
+
+    rng = np.random.RandomState(42)
+    array = rng.rand(10, 100)
+
+    array_with_nans = array.copy()
+    array_with_nans[rng.rand(*array_with_nans.shape) < 0.5] = np.nan
```

Comment:
```suggestion
    array_with_nans = array.copy()
    array_with_nans[rng.rand(*array_with_nans.shape) < 0.5] = np.nan
```

and the replace all occurrences of `array.shape` with `array_with_nans.shape` instead since the unmasked values of `array` are never used in this test.